chall10/pcuadrad.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+import sys
+
+lettersValid = "ABCDEFG"
+
+def checker(lstInput: list) -> bool:
+    '''Check the input user That the box is between A-G,
+        that the second parameter after '_' is capitalized and 
+        there are only two and they alternate '''
+
+    if len(lstInput) > 42:
+        return False
+    for elem, i in zip(lstInput, range(len(lstInput))):
+        if str(elem).find('_') == -1:
+            return False
+        newcolor = str(elem).split('_')
+        if newcolor[0] not in lettersValid or not newcolor[1].istitle():
+            return False
+        if i % 2 == 0:
+            if i > 0 and (fcolor[1] != newcolor[1] or newcolor[1] == scolor[1]):
+                return False
+            fcolor = newcolor
+        else:
+            if i > 1 and (scolor[1] != newcolor[1] or newcolor[1] == fcolor[1]):
+                return False
+            scolor = newcolor
```

Comment:
Assigned but never used?

Reply:
I use it in the conditions, comparing it with the previous fcolor or scolor simultaneously to check that they are not repeated.